uri-handler-sample/src/extension.ts

```diff
@@ -0,0 +1,34 @@
+import * as vscode from 'vscode';
+
+// Our implementation of a UriHandler.
+class MyUriHandler implements vscode.UriHandler {
+	// This function will get run when something redirects to VS Code
+	// with your extension id as the authority.
+	handleUri(uri: vscode.Uri): vscode.ProviderResult<void> {
+		let message = "Handled a Uri!";
+		if (uri.query) {
+			message += ` It came with this query: ${uri.query}`;
+		}
+		vscode.window.showInformationMessage(message);
+	}
+}
+
+export function activate(context: vscode.ExtensionContext) {
+
+	let disposable = vscode.commands.registerCommand('uri-handler-sample.start', async () => {
+		// Create our new UriHandler
+		const uriHandler = new MyUriHandler();
+
+		// And register it with VS Code. You can only register a single UriHandler for your extension.
+		context.subscriptions.push(vscode.window.registerUriHandler(uriHandler));
+
+		// This gives you the exact Uri to put in your browser. The `vscode.env.asExternalUri` will add a
+		// query parameter (ex: "windowId%3D14") that will help VS Code decide which window to redirect to.
```

Comment:
This makes it sound like it's necessary to use `asExternalUri`, when it really isn't. Code will just pick the last focused window, if there's more than 1 open window. Code will also launch a new window if no windows are open but Code is running (macOS only).

So: if there's no requirement to actually go to the same window, I would avoid the reference to `asExternalUri`.

Reply:
The state I was referring to is: Code is running but no windows are open. This state is only possible on macOS. When the last window closes on Windows/Linux, Code exits. But on macOS Code keeps on truckin'.

Reply:
```suggestion
		// You don't have to get the Uri from the `vscode.env.asExternalUri` API but it will add a query
		// parameter (ex: "windowId%3D14") that will help VS Code decide which window to redirect to.
		// If this query parameter isn't specified, VS Code will pick the last windows that was focused.
```